cpp/src/parquet/arrow/reader.cc

```diff
@@ -1315,15 +1315,15 @@ Status FileReader::GetRecordBatchReader(const std::vector<int>& row_group_indice
   return Status::OK();
 }
 
-Status FileReader::Make(::arrow::MemoryPool* pool,
+Result<std::unique_ptr<FileReader>> Make(::arrow::MemoryPool* pool,
                         std::unique_ptr<ParquetFileReader> reader,
                         const ArrowReaderProperties& properties,
                         std::unique_ptr<FileReader>* out) {
   *out = std::make_unique<FileReaderImpl>(pool, std::move(reader), properties);
   return static_cast<FileReaderImpl*>(out->get())->Init();
 }
 
-Status FileReader::Make(::arrow::MemoryPool* pool,
+Result<std::unique_ptr<FileReader>> Make(::arrow::MemoryPool* pool,
```

Comment:
You still have to provide the implementations for `Status FileReader` because they are part of the headers file. Even though they are deprecated you can't remove them.

Reply:
Done! keeping the implementations for `Status FileReader `